agent/engine/execcmdagent/manager.go

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+package execcmdagent
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	apitask "github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+)
+
+const (
+	defaultStartRetryTimeout   = time.Minute * 10
+	defaultRetryMinDelay       = time.Second * 1
+	defaultRetryMaxDelay       = time.Second * 30
+	defaultInspectRetryTimeout = time.Minute * 2
+	maxRetries                 = 5
+	retryDelayMultiplier       = 1.5
+	retryJitterMultiplier      = 0.2
+)
+const (
+	Restarted RestartStatus = iota
+	NotRestarted
+	Unknown
```

Comment:
We might be in `Unknown` land for longer durations (ex. `dockerTimeoutErr` which you reference in your test).  This makes me ask what we will report when we have `RUNNING`|`PENDING`|`STOPPED` to report.

What does the ECS Agent do for `Unknown` -- are we following that pattern here?
I see the point of it existing, but it brings up a little bit of a tangle.

Reply:
What would be the alternative? Ignore?

Reply:
I think agent doesn't update status until it is in a known state.  So we need to find a way of absolutely limiting our time spent in unknown-land.